src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,114 @@
+            }
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(INDEX_ID, Long.class);
+                book.setId(id);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while creating book", e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Connection connection = ConnectionConfig.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_BOOK_BY_ID)) {
+            preparedStatement.setLong(INDEX_ID, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getBook(resultSet));
+            } else {
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error can't get book by ID", e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Connection connection = ConnectionConfig.getConnection();
+             PreparedStatement statement = connection.prepareStatement(SELECT_ALL_BOOKS)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> result = new ArrayList<>();
+            while (resultSet.next()) {
+                result.add(getBook(resultSet));
+            }
+            return result;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionConfig.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_BOOK)) {
+            preparedStatement.setString(TITLE_PART, book.getTitle());
+            preparedStatement.setBigDecimal(PRICE_PART, book.getPrice());
+            preparedStatement.setLong(ID_PART, book.getId());
+            if (preparedStatement.executeUpdate() == 0) {
+                throw new RuntimeException("Updating book failed, no rows affected.");
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Error while updating book", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionConfig.getConnection();
+             PreparedStatement statement = connection.prepareStatement(DELETE_BOOK)) {
+            statement.setLong(INDEX_ID, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with id = " + id, e);
+        }
+    }
+
+    private Book getBook(ResultSet resultSet) {

```

Comment:
I would recommend you chose a more appropriate naming for this method because it looks like parsing or mapping, you have got already a book from db